### PR TITLE
SONARIAC-752 Log filename when file identifier does not match in YamlSensor

### PR DESCRIPTION
**Pull request by nils-werner-sonarsource**

### CODE REVIEW

File: iac-common/src/main/java/org/sonar/iac/common/extension/IacSensor.java [hunk 1]
```diff
@@ -115,6 +115,13 @@ public void execute(SensorContext sensorContext) {
     statistics.log();
   }
 
+  private List<InputFile> inputFiles(SensorContext sensorContext) {
+    FileSystem fileSystem = sensorContext.fileSystem();
+    FilePredicate predicate = mainFilePredicate(sensorContext);
+    return StreamSupport.stream(fileSystem.inputFiles(predicate).spliterator(), false)
+      .collect(Collectors.toList());
+  }
```
Comment (nils-werner-sonarsource) [inline iac-common/src/main/java/org/sonar/iac/common/extension/IacSensor.java:123]: That way the predicates are only applied once. Before there were applied in lines 101 and 158.

File: iac-extensions/cloudformation/src/test/java/org/sonar/iac/cloudformation/plugin/CloudformationSensorTest.java [hunk 2]
```diff
@@ -35,6 +37,8 @@
 
 class CloudformationSensorTest extends ExtensionSensorTest {
 
+  @RegisterExtension
+  LogTesterJUnit5 logTester = new LogTesterJUnit5();
```
Comment (jonas-wielage-sonarsource) [inline iac-extensions/cloudformation/src/test/java/org/sonar/iac/cloudformation/plugin/CloudformationSensorTest.java:41]: Should not be needed, as there is already a logtester in a parent class

File: iac-extensions/kubernetes/src/test/java/org/sonar/iac/kubernetes/plugin/KubernetesSensorTest.java [hunk 3]
```diff
@@ -66,6 +66,11 @@ void shouldNotParseYamlFileWithoutIdentifiers() {
   void shouldNotParseYamlFileWithIncompleteIdentifiers() {
     analyse(sensor(), inputFile("apiVersion: ~\nkind: ~\nmetadata: ~\n"));
     asserNotSourceFileIsParsed();
+
+    var logs = logTester.logs(LoggerLevel.DEBUG);
```
Comment (jonas-wielage-sonarsource) [inline iac-extensions/kubernetes/src/test/java/org/sonar/iac/kubernetes/plugin/KubernetesSensorTest.java:70]: The logtester could also be retrieved from the parent class

**Review by jonas-wielage-sonarsource — APPROVED** [review]
LGTM, only small nitpicks

File: iac-common/src/main/java/org/sonar/iac/common/extension/IacSensor.java (hunk 1 above)
Comment (jonas-wielage-sonarsource) [reply iac-common/src/main/java/org/sonar/iac/common/extension/IacSensor.java:123]: I don't see any duplicate applications before, but looks good nonetheless!

**Pull request merged**